packages/components/src/public/QueryModel/TabbedGridPanel.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2019 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { FC, memo, useCallback, useState } from 'react';
+import classNames from 'classnames';
+
+import { QueryModel } from './QueryModel';
+import { InjectedQueryModels } from './withQueryModels';
+import { GridPanel, GridPanelProps } from './GridPanel';
+
+interface GridTabProps {
+    isActive: boolean;
+    model: QueryModel;
+    onSelect: (id: string) => void;
+    pullRight: boolean;
+}
+
+const GridTab: FC<GridTabProps> = memo(({ isActive, model, onSelect, pullRight }) => {
+    const { id, queryInfo, title } = model;
+    const className = classNames({
+        active: isActive,
+        'pull-right': pullRight,
+    });
+    const onClick = useCallback(() => onSelect(id), [id, onSelect]);
+
+    return (
+        <li className={className}>
+            <a onClick={onClick}>{title || queryInfo.queryLabel || queryInfo.name}</a>
```

Comment:
Any reason to think that `queryInfo` would be undefined at any point, maybe while the models are initializing?

Reply:
If you pass a queryConfig to withQueryModels you are guaranteed a model 100% of the time, so no. If it's the case that you're adding a model later, then yes in theory you may be missing a model during a render cycle. Adding a model later is really not the normal use case, I think the onus is on the consumer of TabbedGridPanel to make sure they're only passing a tabOrder that has a model for every entry.